ui/src/app/edge/live/common/grid/modal/modal.ts

```diff
@@ -1,53 +1,146 @@
 import { Component } from '@angular/core';
+import { FormGroup } from '@angular/forms';
+import { FormlyFieldConfig } from '@ngx-formly/core';
+import { TranslateService } from '@ngx-translate/core';
+import { filter } from 'rxjs/operators';
 import { AbstractModal } from 'src/app/shared/genericComponents/modal/abstractModal';
-import { ChannelAddress, CurrentData, EdgeConfig, GridMode } from 'src/app/shared/shared';
+import { TextIndentation } from 'src/app/shared/genericComponents/modal/modal-line/modal-line';
+import { ChannelAddress, Edge, EdgeConfig, GridMode, Utils } from 'src/app/shared/shared';
+import { Role } from 'src/app/shared/type/role';
+
+export type FormlyFieldLine = {
+  type: string,
+  channel?: string,
+  filter?: Function,
+  converter?: Function,
+  name?: string,
+  nameSuffix?: Function,
```

Comment:
I still don't like `nameSuffix` too much, because it is not easily understandable. Maybe we should either accept a name or a callback and not both. E.g. having a `name` and `nameProvider` instead, or having name of type `string | Function`